include/onnxruntime/interface/provider/provider.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <functional>
+
+#include "interface/graph/graph.h"
+#include "interface/framework/kernel.h"
+//#include "core/session/onnxruntime_c_api.h"
+#include "core/framework/ortdevice.h"
+#include "core/framework/stream_handles.h"
+#include "core/framework/node_compute_info.h"
+#include <climits>
+
+namespace onnxruntime {
+
+namespace interface {
+
+struct SubGraphDef {
+  struct MetaDef {
+    std::string name;
+    std::string domain;
+    int since_version;
+
+    std::vector<std::string> inputs;
+    std::vector<std::string> outputs;
+    std::vector<std::string> constant_initializers;
+    std::string doc_string;
+  };
+
+  std::vector<size_t> nodes;
+  void SetMetaDef(std::unique_ptr<MetaDef>&& meta_def) { meta_def_ = std::move(meta_def); }
+  MetaDef* GetMetaDef() { return meta_def_.get(); };
+
+ private:
+  std::unique_ptr<MetaDef> meta_def_;
+};
+
+struct Allocator {
+  virtual ~Allocator() = default;
+  enum DevType {
+    CPU = 0,
+    GPU,
+    FPGA,
+    TPU,
+  };
```

Comment:
should we decouple the existing device type definition from ORT headers and reuse it here?

and btw, what is TPU?